### PR TITLE
Add audit script to list commits from Spark

### PR DESCRIPTION
**Pull request by nartal1**
Adding audit script which is used to get the commits from Apache Spark. Currently we are auditing Spark-3.2 which is the master branch in Spark. New scripts can be added/modified if we want to track and get the list for other branches as well. 
This fixes https://github.com/NVIDIA/spark-rapids/issues/1136

Signed-off-by: Niranjan Artal <nartal@nvidia.com>

### CODE REVIEW

**Comment by nartal1** [comment]
build

File: scripts/audit-spark-3.2.sh [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  ## Spark release works in this way -  Once the release branch is cut, PR's are merged into master and then cherry-picked to release branches.
+  ## This causes different commit ids for the same PR in different branches(master & release branch).
+  ## We need to find the common parent before branch-3.1 was cut. In this case commit id 990bee9c58e is the one.
+  ## So we get all commits from master and branch-3.1 until the common parent commit and then filter it based on commit header message i.e
+  ## if the commit header is same, it means it is cherry-picked to branch-3.1(implying that commit is already audited).
+  echo "file $lastcommit not found"
+  cd ${SPARK_TREE}
+
+  ## Get all the commits from TOT tagv3.1.1-rc3 to 990bee9c58e
+  git checkout v3.1.1-rc3
```
Comment (razajafri) [inline scripts/audit-spark-3.2.sh:76]: This can be a command line option

File: scripts/audit-spark-3.2.sh [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+REF=main
+while getopts v:a: flag
+do
+  case "${flag}" in
+      v) lastcommit=${OPTARG};;
+      b) basebranch=${OPTARG};;
+  esac
+done
+
+SPARK_TREE="$HOME/spark-3.2/spark"
```
Comment (razajafri) [inline scripts/audit-spark-3.2.sh:42]: can this just be pulled from `$SPARK_HOME`??

File: scripts/audit-spark-3.2.sh (hunk 1 above)
Comment (nartal1) [reply scripts/audit-spark-3.2.sh]: Thanks! changed it to command line option.

File: scripts/audit-spark-3.2.sh (hunk 2 above)
Comment (nartal1) [reply scripts/audit-spark-3.2.sh]: Agreed. Updated it to $SPARK_HOME

**Review by razajafri — APPROVED** [review]

**Comment by nartal1** [comment]
build

**Pull request merged**